client/web/src/insights/pages/search-insight-creation/components/form-series-input/FormSeriesInput.tsx

```diff
@@ -0,0 +1,114 @@
+import classnames from 'classnames'
+import React from 'react'
+import { noop } from 'rxjs'
+
+import { useField } from '../../hooks/useField'
+import { useForm } from '../../hooks/useForm'
+import { DataSeries } from '../../types'
+import { DEFAULT_ACTIVE_COLOR, FormColorInput } from '../form-color-input/FormColorInput'
+import { InputField } from '../form-field/FormField'
+import { createRequiredValidator } from '../validators'
+
+const requiredNameField = createRequiredValidator('Name is required field for data series.')
+const validQuery = createRequiredValidator('Query is required field for data series.')
+
+interface FormSeriesProps {
+    /** Name of series. */
+    name?: string
+    /** Query value of series. */
+    query?: string
+    /** Color value for line chart. (series) */
+    color?: string
+    /** Enable autofocus behavior of first input of form. */
+    autofocus?: boolean
+    /** Enable cancel button. */
+    cancel?: boolean
+    /** Custom class name for root element of form series. */
+    className?: string
+    /** On submit handler of series form. */
+    onSubmit?: (series: DataSeries) => void
+    /** On cancel handler. */
+    onCancel?: () => void
+}
+
+/** Displays form series input (three field - name field, query field and color picker). */
+export const FormSeriesInput: React.FunctionComponent<FormSeriesProps> = props => {
+    const { name, query, color, className, cancel = false, autofocus = true, onCancel = noop, onSubmit = noop } = props
+
+    const hasNameControlledValue = !!name
+    const hasQueryControlledValue = !!query
+
+    const { formAPI, handleSubmit, ref } = useForm({
+        initialValues: {
+            seriesName: name ?? '',
+            seriesQuery: query ?? '',
+            seriesColor: color ?? DEFAULT_ACTIVE_COLOR,
+        },
+        onSubmit: values =>
+            onSubmit({
+                name: values.seriesName,
+                query: values.seriesQuery,
+                color: values.seriesColor,
+            }),
+    })
+
+    const nameField = useField('seriesName', formAPI, requiredNameField)
+    const queryField = useField('seriesQuery', formAPI, validQuery)
+    const colorField = useField('seriesColor', formAPI)
+
+    return (
+        <div ref={ref} className={classnames('d-flex flex-column', className)}>
+            <InputField
+                title="Name"
+                required={true}
+                autoFocus={autofocus}
+                placeholder="Example: Function component"
+                description="Name shown in the legend and tooltip"
+                valid={(hasNameControlledValue || nameField.meta.touched) && nameField.meta.validState === 'VALID'}
+                error={nameField.meta.touched && nameField.meta.error}
+                {...nameField.input}
+            />
+
+            <InputField
+                title="Query"
```

Comment:
```suggestion
                title="Search query"
```

Wondering if we should explicitly call it a "search query" since query means other things in general graph data (sql query, graphlql query, etc) cc @AlicjaSuska